saleor/graphql/account/mutations/staff/customer_update.py

```diff
@@ -162,3 +166,22 @@ def perform_mutation(cls, _root, info: ResolveInfo, /, **data):
 
         # Return the response
         return cls.success_response(new_instance)
+
+    @classmethod
+    @traced_atomic_transaction()
+    def save(cls, info: ResolveInfo, instance, cleaned_input):
+        manager = get_plugin_manager_promise(info.context).get()
+
+        default_addresses = cls.save_and_apply_addresses_from_input(
+            cleaned_input=cleaned_input, user_instance=instance
+        )
+
+        if default_addresses["billing"]:
+            instance.addresses.add(default_addresses["billing"])
+        if default_addresses["shipping"]:
+            instance.addresses.add(default_addresses["shipping"])
+
+        instance.search_document = prepare_user_search_document_value(instance)
+        instance.save()
```

Comment:
Could we save only updated fields instead of the whole instance? If it will extend the scope too much, we can create a ticket for that

Reply:
I think it's not an easy way to deduct that quickly and it's not the scope of my refactor (I'm breaking inheritance, not optimizing writes) - so I will create a ticket

Reply:
https://linear.app/saleor/issue/EXT-2149/optimize-save-operations-in-customeraccount-createupdate